ttnn/cpp/ttnn/operations/experimental/ssm/prefix_scan/device/prefix_scan.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "prefix_scan.hpp"
+
+#include "prefix_scan_program_factory.hpp"
+
+namespace ttnn::operations::experimental::ssm {
+
+void PrefixScan::validate(const std::vector<Tensor>& input_tensors) const {
+    TT_FATAL(input_tensors.size() == 3, "Expected 3 input tensors (A, Bx, H)");
+
+    const auto& a = input_tensors.at(0);
```

Comment:
good example: in lime 12 we already have a check so simply using input_tensors[0] will do the same.